Sources/Plasma/PubUtilLib/plGImage/plPNG.cpp

```diff
@@ -226,27 +229,72 @@ bool plPNG::IWrite(plMipmap* source, hsStream* outStream)
             row_ptrs[i] = (png_bytep)srcp + (i * stride);
         }
 
+        // Write image data
         png_write_image(png_ptr, row_ptrs);
+
+        // Prepare Textual Metadata
+        char time_string[26]; // RFC-1123: "Day, DD-Mon-YYYY hh:mm:ss";
+        time_t rawtime = time(nullptr);
+        strftime(time_string, sizeof(time_string), "%a, %d-%b-%Y %T", gmtime(&rawtime));
+        std::multimap<ST::string, ST::string> standard_fields;    
+        standard_fields.emplace("Software", plProduct::ProductString());
+        standard_fields.emplace("Creation Time", time_string);
+
+        png_text* text = new png_text[standard_fields.size() + textFields.size()];
+        auto text_buffer = new std::vector<char*>();
+        text_buffer->reserve(standard_fields.size() + textFields.size());
+        int num_txtfields = 0;
+        for (auto it = standard_fields.begin(); it != standard_fields.end(); it++, num_txtfields++) {
+            // The PNG specification requires Latin-1 in these fields
+            auto key_text = strdup(it->first.left(79).to_latin_1().data());
+            auto val_text = strdup(it->second.to_latin_1());
+            text[num_txtfields].key = const_cast<png_charp>(key_text);
+            text[num_txtfields].text = const_cast<png_charp>(val_text);
+            text[num_txtfields].compression = PNG_TEXT_COMPRESSION_NONE;
+            text_buffer->push_back(key_text);
+            text_buffer->push_back(val_text);
+        }
```

Comment:
This could all be simplified by creating a single multimap which copies `textFields` and merges in the two standard fields.  That way, you only have one loop to convert.

Reply:
😢 That's how I had it originally, before the rewrites...  

I'll gladly simplify this again.